Improve and clarify host task interop description (#260)
KhronosGroup/SYCL-Docs: adoc/chapters/programming_interface.adoc

    | Access a buffer from a <<host-task>>.
|====

Programs which specify the access target as [code]#target::device# and then

> capture the [code]#accessor# in a <<host-task>> can only use the accessor for
> interoperability through the [code]#interop_handle#.

Reviewer: ```suggestion
capture the [code]#accessor# in a <<host-task>> can use the accessor for
interoperability through the [code]#interop_handle#.  Any other use of of such an
accessor in a <<host-task>> results in undefined behavior.
```

I think we should state explicitly that you get UB if you use a `device` accessor in any other way from a host task.

Please also close internal issue 609 when this is merged.

Author: Updated as suggested to specify that it is undefined behavior.

Programs which specify the access target as [code]#target::host_task# and then

> use the [code]#accessor# from a <<sycl-kernel-function>> are ill formed.

Reviewer: ```suggestion
use the [code]#accessor# from a <<sycl-kernel-function>> result in undefined behavior.
```

It's probably better to  be consistent with the previous paragraph and say this  is UB vs. ill formed.  Does the term "ill formed" mean that the compiler is required to issue a diagnostic?

Since a <<host-task>> is invoked directly by the <<sycl-runtime>> rather
than being compiled as a <<sycl-kernel-function>>, it does not have the same
restrictions as a <<sycl-kernel-function>>, and can therefore contain any
arbitrary {cpp} code.

Capturing <<accessor,accessors>> in a <<host-task>> is allowed, however,
capturing or using any other SYCL class that has reference semantics (see

> <<sec:reference-semantics>>) is undefined behavior.

Reviewer: Thanks.  Please also  close internal issue 605 when this is merged.
